### PR TITLE
add bfd support to backwards-cpp

### PR DESCRIPTION
**Pull request by enzalito**
This PR adds bfd support to backwards-cpp. With it, stack traces show more details about the sources.

### CODE REVIEW

File: packages/b/backward-cpp/xmake.lua [hunk 1]
```diff
@@ -19,6 +25,11 @@ package("backward-cpp")
     add_deps("cmake")
 
     on_install("linux", "mingw", "macosx", "windows", function (package)
+        if package:config("bfd") then
+            package:add("deps", "binutils")
+            package:add("syslinks", "bfd")
+        package:add("cxxflags", "-DBACKWARD_HAS_BFD=1")
```
Comment (star-hengxing) [inline packages/b/backward-cpp/xmake.lua:31]:
```lua
package:add("defines, "")
```

File: packages/b/backward-cpp/xmake.lua [hunk 2]
```diff
@@ -19,6 +25,11 @@ package("backward-cpp")
     add_deps("cmake")
 
     on_install("linux", "mingw", "macosx", "windows", function (package)
+        if package:config("bfd") then
+            package:add("deps", "binutils")
```
Comment (star-hengxing) [inline packages/b/backward-cpp/xmake.lua:29]: We need to dwonload/fetch deps beforce on_install. Move `package:add("deps", "binutils")` to on_load

File: packages/b/backward-cpp/xmake.lua [hunk 3]
```diff
@@ -19,6 +25,11 @@ package("backward-cpp")
     add_deps("cmake")
 
     on_install("linux", "mingw", "macosx", "windows", function (package)
+        if package:config("bfd") then
```
Comment (star-hengxing) [inline packages/b/backward-cpp/xmake.lua:28]: Only linux/macosx/msys support binutils, please exclude other plat

**Comment by enzalito** [comment]
@star-hengxing thank you for the feedback ! I just pushed a new commit.  
I haven't used xmake a whole lot yet so I might not have made the changes you envisioned, I'll happily fix my PR if that is the case.

File: packages/b/backward-cpp/xmake.lua [hunk 4]
```diff
@@ -24,11 +24,16 @@ package("backward-cpp")
...
         if package:config("bfd") then
             package:add("deps", "binutils")
+        end
+    end)
+
+    on_install("linux", "mingw", "macosx", "windows", function (package)
+        if package:config("bfd") and package:is_plat("linux", "mingw@msys", "macos") then
             package:add("syslinks", "bfd")
-        package:add("cxxflags", "-DBACKWARD_HAS_BFD=1")
+            package:add("defines", "-DBACKWARD_HAS_BFD=1")
```
Comment (waruqi) [inline packages/b/backward-cpp/xmake.lua:36]: remove `-D`

**Comment by enzalito** [comment]
@waruqi could you merge that please ?

**Pull request merged**